File: c/Prata_languageC/ch06/ex14.c
// Copyright (C) Khudyashev Ivan, 2018-2020, bahek1983@gmail.com
#include <stdio.h>

int main(void)
{
    
    printf("Start Program\n\n");
    // program here
    const int ARRAY_LEN = 8;
    double numbers[ARRAY_LEN];
    double sumNumbers[ARRAY_LEN];
    const char msg[] = "Input %d numbers separated by space: ";
    const char formatStr = "%10.2f ";
    printf(msg, ARRAY_LEN);
    for(int i = 0; i < ARRAY_LEN; i++) {
        scanf("%f", numbers + i);
    }
    // suppose that first array consist at least one element with index 0
    sumNumbers[0] = numbers[0];
    for(int i = 1; i < ARRAY_LEN; i++) {
        sumNumbers[i] = sumNumbers[i - 1] + numbers[i];
    }
    for(int i = 0; i < ARRAY_LEN; i++) {
        printf(formatStr, numbers[i]);
    }
    for(int i = 0; i < ARRAY_LEN; i++) {
        printf(formatStr, sumNumbers[i]);
    }
    printf("\nFinish Program!\n");
    return 0;
}

File: python3/ThePythonWorkshop/ch01/ex14.py
# Copyright (C) 2019-2021, Khudyashev Ivan, bahek1983@gmail.com

def doAction():
    age = 20
    print("age < 13: ", age < 13)
    print("age >= 20 and age <= 21: ", age >=20 and age <= 21 )
    print("age != 21: ", age != 21)
    print("age == 19: ", age == 19)
    print("6 == 6.0: ", 6 == 6.0)
    print("6 == '6': ", 6 == '6')
    print("(age >= 20 and age < 30) or (age >= 30 and age < 40): ", \
        (age >= 20 and age < 30) or (age >= 30 and age < 40) \
    )
    print("(20 <= age < 30) or (30 <= age < 40): ",  \
        (20 <= age < 30) or (30 <= age < 40) \
    )

def main():
    doAction()

if __name__ == "__main__":
    main()
